web/src/utilities/repeater.js

```diff
@@ -0,0 +1,28 @@
+export class Repeater {
+    constructor(handlerFunc, sleepMs) {
+        this.handlerFunc = handlerFunc;
+        this.sleepMs = sleepMs;
+        this.stop = false;
+    }
+
+    Start = () => {
+        this.repeat();
+        return this.cancel;
+    }
+
+    cancel = () => {
+        this.stop = true;
+    }
+
+    repeat = () => {
+        this.handlerFunc().then(() => {
```

Comment:
you can use .finally here instead

Reply:
@salahalsaleh1993 do you mean instead of `then` or instead of both `then` and `catch`?

Reply:
instead of both, actually don't mind that, `.catch` is important in this case to not interrupt the workflow